src/Compilers/Core/Portable/RealParser.cs

```diff
@@ -56,20 +47,12 @@ public static bool TryParseDouble(string s, out double d)
         /// <returns>True of the input was converted; false if there was an overflow</returns>
         public static bool TryParseFloat(string s, out float f)
         {
-            try
-            {
-                var str = DecimalFloatingPointString.FromSource(s);
-                var dbl = FloatFloatingPointType.Instance;
-                ulong result;
-                var status = RealParser.ConvertDecimalToFloatingPointBits(str, dbl, out result);
-                f = Int32BitsToFloat((uint)result);
-                return status != Status.Overflow;
-            }
-            catch (System.FormatException)
-            {
-                f = 0.0f;
-                return false;
-            }
```

Comment:
Want to keep the `try / catch` here until we have more extensive testing on this code base.  The parsing not handling error cases sufficiently lead to the top reported issue on the compiler for Update 1 RC.  The `try / catch` is a good safeguard against that.